test/runner/runner.go

```diff
@@ -26,22 +26,58 @@ import (
 func NewRegistry() *TestRegistry {
 	return &TestRegistry{
 		tests: make(map[string]Test),
+		testGroups: make(map[string]TestGroup),
 	}
 }
 
-// TestRegistry contains a mapping of named tests
+//NewTestGroup returns a pointer to a new TestGroup
+func NewTestGroup(name string) *TestGroup {
+	return &TestGroup{
+		name: name,
+		tests: make(map[string]Test),
+	}
+}
+
+// Test is a test function
+type Test func(t *testing.T)
+
+
+// TestRegistry contains a mapping of named test groups
 type TestRegistry struct {
 	suite.SetupTestSuite
 	tests map[string]Test
+	testGroups map[string]TestGroup
 }
 
-// Register registers a named test
+//TestGroup to run multiple tests
+type TestGroup struct {
+	name string
+	suite.SetupTestSuite
```

Comment:
I actually don't think the use of `suite` is needed. We should remove the usage of these interfaces as we're not even running them as test suites.